src/cplus_plugin/models/constant_raster.py

```diff
@@ -0,0 +1,451 @@
+# -*- coding: utf-8 -*-
+"""Models for Constant Raster according to architectural specification."""
+
+from __future__ import annotations
+import os
+import sys
+import dataclasses
+import typing
+from datetime import datetime
+from typing import NamedTuple
+from qgis.core import QgsRasterLayer
+
+from .base import LayerModelComponent, ModelComponentType
+from ..definitions.constants import (
+    PATH_ATTRIBUTE,
+    COMPONENT_UUID_ATTRIBUTE,
+    COMPONENT_ID_ATTRIBUTE,
+    COMPONENT_TYPE_ATTRIBUTE,
+    SKIP_RASTER_ATTRIBUTE,
+    ENABLED_ATTRIBUTE,
+    VALUE_INFO_ATTRIBUTE,
+    NORMALIZED_ATTRIBUTE,
+    ABSOLUTE_ATTRIBUTE,
+    MIN_VALUE_ATTRIBUTE_KEY,
+    MAX_VALUE_ATTRIBUTE_KEY,
+    ALLOWABLE_MIN_ATTRIBUTE,
+    ALLOWABLE_MAX_ATTRIBUTE,
+    LAST_UPDATED_ATTRIBUTE,
+    COMPONENTS_ATTRIBUTE,
+    DISPLAY_NAME_ATTRIBUTE,
+    RASTER_COLLECTION_ATTRIBUTE,
+    INPUT_RANGE_ATTRIBUTE,
+)
+
+
+class InputRange(NamedTuple):
+    """Range for input values with named min/max fields for readability."""
+
+    min: float
+    max: float
+
+
+@dataclasses.dataclass
+class ConstantRasterInfo:
+    """Value information for a constant raster.
+
+    This dataclass only handles value information (normalized and absolute values),
+    File/layer information should be handled separately by ConstantRasterComponent.
+    """
+
+    normalized: float = 0.0  # Normalized value (0-1 range)
+    absolute: float = 0.0  # Absolute constant value for creating rasters
+
+    def to_dict(self) -> dict:
```

Comment:
@osundwajeff we move the from/to dict to helpers. Applicable in other data classes.